core/src/main/java/org/infinispan/xsite/ClusteredCacheBackupReceiver.java

```diff
@@ -257,6 +260,25 @@ private XSiteStatePushCommand newStatePushCommand(List<XSiteState> stateList) {
       return defaultHandler.cache().clearAsync();
    }
 
+   @Override
+   public CompletionStage<Boolean> touchEntry(Object key) {
+      int segment = clusteringDependentLogic.getCacheTopology().getSegment(key);
```

Comment:
shouldn't be sent to the primary owner?

Reply:
Hrmm, yeah this should invoke the touch command. Let me do that.